Casks/g/gretl.rb

```diff
@@ -11,9 +11,8 @@
   homepage "https://gretl.sourceforge.net/"
 
   livecheck do
-    url "https://gretl.sourceforge.net/osx.html"
-    regex(/gretl[._-]v?(\d+\w)[._-]macos[._-]#{arch}\.pkg/i)
-    strategy :page_match
+    url "https://sourceforge.net/projects/gretl/rss?path=/gretl"
+    regex(%r{url=.*?/gretl[._-]v?(\d+\w)[._-]macos[._-]#{arch}\.pkg}i)
```

Comment:
```suggestion
    regex(%r{url=.*?/gretl[._-]v?(\d+(?:\.\d*)+[a-z]?)[._-]macos[._-]#{arch}\.pkg}i)
```

We're matching versions like `1.10.2` or `2022c`, so we should use the `*` form of the standard regex (which handles the numeric part of both).

There's also something to be said for using `[a-z]` instead of `\w` here. We only intend to match an optional letter, but `\w` also matches `0-9` and `-`, so it doesn't accurately represent our intention.